### PR TITLE
Add provider documentation

### PR DESCRIPTION
**Pull request by marcofranssen**
- Generate docs for provider
- Improve provider documentation
- Improve Dex oauth2 client resource docs

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+description: |-
+  Provision Dex resources via the Dex gRPC API.
+---
+
+# dexidp Provider
+
+Provision Dex resources via the Dex gRPC API.
+
+## Example Usage
+
```
Comment (truongnht) [inline docs/index.md:14]: Make sure that Dex server is running with grpc api reachable via 127.0.0.1:5557

File: pkg/dexidp/provider.go [hunk 2]
```diff
@@ -31,9 +31,11 @@ func (p *dexProvider) Metadata(ctx context.Context, req provider.MetadataRequest
 // Schema defines the provider-level schema for configuration data.
 func (p *dexProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
 	resp.Schema = schema.Schema{
+		Description: "Provision Dex resources via the Dex gRPC API.",
 		Attributes: map[string]schema.Attribute{
 			"host": schema.StringAttribute{
```
Comment (truongnht) [inline pkg/dexidp/provider.go:36]: since the description mentions host and port, I am thinking about renaming this attribute to address.

**Review by truongnht — COMMENTED** [review]
some minor comments

**Review by truongnht — COMMENTED** [review]
some minor comments

File: docs/index.md (hunk 1 above)
Comment (marcofranssen) [reply docs/index.md:14]: This is generated.

**Review by truongnht — APPROVED** [review]

File: pkg/dexidp/provider.go (hunk 2 above)
Comment (marcofranssen) [reply pkg/dexidp/provider.go:36]: as dex gRPC api also names this host I think we should keep that naming.

File: docs/index.md (hunk 1 above)
Comment (marcofranssen) [reply docs/index.md:14]: Updated it in the example by adding a comment to terraform

**Pull request merged**